src/base/searchengine.cpp

```diff
@@ -104,19 +104,26 @@ QStringList SearchEngine::enabledPlugins() const
     return plugins;
 }
 
-QStringList SearchEngine::supportedCategories() const
+QStringList SearchEngine::supportedCategories(const QString &name) const
 {
-    QStringList result;
-    foreach (const PluginInfo *plugin, m_plugins.values()) {
-        if (plugin->enabled) {
-            foreach (QString cat, plugin->supportedCategories) {
-                if (!result.contains(cat))
-                    result << cat;
-            }
-        }
+    QStringList plugins;
+    if (name == "all")
+        plugins = allPlugins();
+    else if ((name == "enabled") || (name == "multi"))
+        plugins = enabledPlugins();
+    else
+        plugins << name.trimmed();
+
+    QStringList categories;
```

Comment:
I would also change this to `QSet<QString>`

Reply:
Updated to use a `QSet` but still return a `QStringList`